src/Compilers/CSharp/Portable/Symbols/RangeVariableSymbol.cs

```diff
@@ -61,9 +59,11 @@ public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
         {
             get
             {
-                SyntaxToken token = (SyntaxToken)_locations[0].SourceTree.GetRoot().FindToken(_locations[0].SourceSpan.Start);
+                var location = _locations[0];
```

Comment:
Related to this file but not this line: Consider adding an assertion in constructor (line 24) for non-nullable parameters.

Reply:
Now added